### PR TITLE
sys-apps/firejail: Version bump, disables overlayfs to fix privesc

### PR DESCRIPTION
**Pull request by hlein**
New version disables overlayfs, which has a root privesc vuln.
Some new profiles and other minor fixes also included. Disable
overlayfs USE flag in live ebuild as well.

Signed-off-by: Hank Leininger <hlein@korelogic.com>
Closes: https://bugs.gentoo.org/769230
Bug: https://bugs.gentoo.org/769542
Package-Manager: Portage-3.0.14, Repoman-3.0.2

### CODE REVIEW

**Comment by hlein** [comment]
> New issues caused by PR:
> https://qa-reports.gentoo.org/output/gentoo-ci/59eb0489c3/output.html#dev-lang/lfe

I can't see how this has anything to do with this PR - dev-lang/lfe has nothing to do with firejail. Is qa-reports misfiring, or flagging something else? It looks like dev-lang/lfe was just touched with commit https://github.com/gentoo/gentoo/commit/1347cc863ef59c588aabbe27d7cb9067801cb004

**Comment by thesamesam** [comment]
We don't need to disable overlayfs if we're doing the bump, right?

**Comment by thesamesam** [comment]
(ignore closure, pressed wrong button, sorry!)

**Comment by hlein** [comment]
> We don't need to disable overlayfs if we're doing the bump, right?

The fine print of the version bump is that it removes overlayfs support (for now?); see https://github.com/netblue30/firejail/releases/tag/0.9.64.4

So a USE flag for it is meaningless / misleading.

**Comment by thesamesam** [comment]
Serves me right for not checking ;)

**Comment by hlein** [comment]
> Serves me right for not checking ;)

Heh no worries =)

**Comment by thesamesam** [comment]
Could you have a look at running tests when Firejail isn't installed? I tried naively setting PATH in src_test but it got upset about "too many environment variables".

```
>>> Test phase: sys-apps/firejail-0.9.64.4
make -j4   test
make -C test profiles
make -C test fcopy
make -C test fs
make[1]: Entering directory '/var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test'
cd profiles && ./profiles.sh 2>&1 | tee profiles.log
make[1]: Entering directory '/var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test'
cd fcopy && ./fcopy.sh 2>&1 | tee fcopy.log
make[1]: Entering directory '/var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test'
cd fs && ./fs.sh 2>&1 | tee fs.log
make -C test utils
make[1]: Entering directory '/var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test'
cd utils && ./utils.sh 2>&1 | tee utils.log
TESTING: profile conditional (test/profiles/conditional.exp)
TESTING: fcopy cmdline (test/fcopy/cmdline.exp)
spawn /bin/bash
TESTING: build (test/utils/build.exp)
spawn /bin/bash
spawn /bin/bash
TESTING: mkdir/mkfile (test/fs/mkdir_mkfile.exp)
firejail --debug --nodbus --profile=cond1.profile
spawn /bin/bash
fcopy
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/profiles $ firejail --debug --nodbus --profile=cond1.profile
firejail --private --profile=mkdir_mkfile.profile
bash: firejail: command not found
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/profiles $ portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fcopy $ fcopy
bash: fcopy: command not found
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fcopy $ portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fs $ firejail --private --profile=mkdir_mkfile.profile
bash: firejail: command not found
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fs $ echo testing > ~/firejail-test-file-7699
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/utils $ echo testing > ~/firejail-test-file-7699
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/utils $ firejail --build cat ~/firejail-test-file-7699
bash: firejail: command not found
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/utils $ TESTING ERROR 0

TESTING: profile recursivity (test/profiles/profile_recursivity.exp)
TESTING ERROR 0

TESTING: fcopy directory (test/fcopy/dircopy.exp)
spawn /bin/bash
TESTING ERROR 0

spawn /bin/bash
firejail --profile=test3.profile
TESTING: read/write (test/fs/read-write.exp)
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/profiles $ firejail --profile=test3.profile
bash: firejail: command not found
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/profiles $ spawn /bin/bash
firejail --read-only=~/_firejail_test_dir --read-write=~/_firejail_test_dir/test1
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fs $ firejail --read-only=~/_firejail_test_dir --read-write=~/_firejail_test_dir/test1
bash: firejail: command not found
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fs $ TESTING ERROR 0

./utils.sh: line 21: faudit: command not found
TESTING: audit (test/utils/audit.exp)
spawn /bin/bash
firejail --audit
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/utils $ firejail --audit
bash: firejail: command not found
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/utils $ rm -fr dest/*
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fcopy $ rm -fr dest/*
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fcopy $ fcopy src dest
bash: fcopy: command not found
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fcopy $ find dest
dest
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fcopy $ find dest
dest
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fcopy $ find dest
dest
portage@localhost /var/tmp/portage/sys-apps/firejail-0.9.64.4/work/firejail-0.9.64.4/test/fcopy $ TESTING ERROR 0
```

**Comment by hlein** [comment]
> Could you have a look at running tests when Firejail isn't installed? I tried naively setting PATH in src_test but it got upset about "too many environment variables".

Hm, sure. OK, regardless of fixing "make tests runnable when firejail is not installed on the system yet", tests are broken again with current Firejails, by that "too many environment variables" issue.

That comes because recent firejail versions impose some sane-looking-at-first-glance limits on the number and size of env vars, but portage exceeds those limits.

I've already reported that upstream; I'll add this to the problems caused: https://github.com/netblue30/firejail/issues/3851

Until/unless I get upstream to implement a solid fix like making the env-limits something profile-driven (and then Gentoo can deploy profiles that raise or remove those limits), I've got workarounds in the works, see https://bugs.gentoo.org/694966

But that list keeps growing, so I'm not ready to make a PR for it. And I bet `make test` will explicitly require some firejail bits that will refuse to run with the environment that portage runs with (like, uh, `firejail`). So I might just end up patching out those limits, as another commenter already said they do.

Actually, I'm not sure if/how firejail tests could ever completely work just out of `$WORKDIR`, because:
- firejail gets installed setuid root
- it needs that setuid-root for full functionality, sandboxing, etc.
- `make test` attempts to exercise that functionality
- in `$WORKDIR`, firejail is only `portage:portage` mode 755
- so running tests on the just-compiled-but-not-installed files wouldn't work

Meanwhile, testing with a setuid firejail (installed, or running from the `image/` dir after `ebuild install` but before `qmerge`, if that is a thing) also has problems, because portage uses `LD_PRELOAD` for `libsandbox.so`, and of course `LD_PRELOAD` isn't allowed when executing a setuid binary. I think those don't actually cause `ebuild ... test` to fail, it just emits various warnings and probably does not behave as an admin would expect (the `too many environment variables` error is indeed fatal for some tests, however).

Is there precedent for other packages in a similar situation? I just looked at a few random packages that provide setuid binaries in `/usr/bin/` and none of them mention tests at all; `ebuild ... test` does nothing meaningful.

tl;dr: it's complicated and tests are going to stay some kind of problematic, can we merge the security fix in the meantime?

**Comment by hlein** [comment]
> can we merge the security fix in the meantime?

Aaay, and now I see you did so. Thanks!
